benchalerts/benchalerts/clients.py

```diff
@@ -0,0 +1,400 @@
+# Copyright (c) 2022, Voltron Data.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import datetime
+import enum
+import os
+import textwrap
+from json import dumps
+from typing import Optional
+
+import jwt
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+from .log import fatal_and_log, log
+
+
+class _BaseClient(abc.ABC):
+    """A client to interact with an API.
```

Comment:
For the future: with which parts in this repo does this overlap?

Reply:
The `benchclients` package was inspired by this code. @alistaire47 and I both needed a python client to hit the Conbench API for `benchadapt` and `benchalerts` respectively, so he created `benchclients` so both packages could depend on the same code.

I still need to do https://github.com/conbench/benchalerts/issues/16 to delete this code and have this package depend on `benchclients` so we can reduce duplication. (I should move that repo's issues to this repo now!)

Reply:
okay, moved all the issues, so this is now https://github.com/conbench/conbench/issues/754